langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -2278,3 +2259,52 @@ function testUsingCloneableReturnType() {
     assertEquality("dummyVal", caller.getAttribute("dummy"));
 }
 
+function testDecimalZeroToStringWithDifferentPrecisions() returns error? {
+    decimal d1 = 0.0d;
+    decimal d2 = 0d;
+
+    assertFalse(d1.toString() == d2.toString());
+    assertTrue(check decimal:fromString(d1.toString()) == check decimal:fromString(d2.toString()));
+    assertFalse(decimal:fromString(d1.toString()) === decimal:fromString(d2.toString()));
+
+    d1 = 0.00000000d;
+    d2 = 0.0d;
+
+    assertFalse(d1.toString() == d2.toString());
+    assertTrue(check decimal:fromString(d1.toString()) == check decimal:fromString(d2.toString()));
```

Comment:
If `check decimal:fromString(d1.toString())` results in the error being returned from this function, the test will not fail. IMO, it's better to handle the error here or use checkpanic.

Reply:
replaced  with `checkpanic` thanx